main.go

```diff
@@ -237,17 +237,20 @@ func startHttpServer(c *config.Config, httpServer **http.Server,
 		WriteTimeout: c.HTTPWriteTimeout,
 	}
 
+	checkClientCertForReads := c.TLSCaFile != ""
 	checkClientCertForWrites := c.TLSCaFile != "" && c.AllowUnauthenticatedReads
```

Comment:
These two lines need adjustment.
* checkClientCertForReads should be false if c.AllowUnauthenticatedReads is true.
* checkClientCertForWrites should be true if c.TLSCaFile is set.